src/VCR/Videorecorder.php

```diff
@@ -95,13 +95,13 @@ public function getEventDispatcher()
     /**
      * Dispatches an event to all registered listeners.
      *
-     * @param string $eventName The name of the event to dispatch.
      * @param Event $event The event to pass to the event handlers/listeners.
+     * @param string $eventName The name of the event to dispatch.
      * @return Event
      */
-    private function dispatch($eventName, Event $event = null)
+    private function dispatch(Event $event, $eventName = null)
     {
-        return $this->getEventDispatcher()->dispatch($eventName, $event);
+        return $this->getEventDispatcher()->dispatch($event, $eventName);
```

Comment:
If VCR is to support LTS Symfony releases, we'll need to check the version at runtime as 4.3 contains a breaking change which affects the order of parameters to the `dispatch()` method of the event dispatcher. The 1.5 branch currently uses some dirty reflection to determine the order. See https://github.com/php-vcr/php-vcr/commit/71372309fc35d22223085fe65d95516cc2e6b19d.

Reply:
I can't find a version constant or something else we can check during runtime to determine the order of arguments, as alternative to the reflection solution. Do you know of an other option?

Reply:
It's probably the most straightforward solution. However, since `Symfony\Components\EventDispatcher\Event` was removed in the same release the signature was changed, we could check for the existence of that class.

Example:
```php
private function dispatch(Event $event, $eventName = null)
{
    if (class_exists('Symfony\Components\EventDispatcher\Event') {
        return $this->getEventDispatcher()->dispatch($event, $eventName);
    }
    
    return $this->getEventDispatcher()->dispatch($eventName, $event);
}
```

Changelog: https://github.com/symfony/symfony/blob/678e066aa8b2d91bec50c56a081435300f58fcf8/src/Symfony/Component/EventDispatcher/CHANGELOG.md#500